client/containers/Pub/PubDocument/PubFileImport/MetadataEditor.js

```diff
@@ -0,0 +1,275 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Button, Checkbox, InputGroup, TextArea } from '@blueprintjs/core';
+
+import { apiFetch } from 'utils';
+import { usePageContext } from 'utils/hooks';
+import { Avatar } from 'components';
+import { MenuButton, MenuItem } from 'components/Menu';
+import { profileUrl } from 'shared/utils/canonicalUrls';
+import { usePubContext } from '../../pubHooks';
+
+require('./metadataEditor.scss');
+
+const attributionShape = PropTypes.shape({
+	name: PropTypes.string,
+	users: PropTypes.arrayOf(PropTypes.shape({})),
+});
+
+const propTypes = {
+	onSetMetadataUpdater: PropTypes.func.isRequired,
+	proposedMetadata: PropTypes.shape({
+		attributions: PropTypes.arrayOf(attributionShape),
+	}).isRequired,
+};
+
+const guessInitials = (name) => {
```

Comment:
It'd be nice if we broke out the firstName/lastName split and initial generation into a single util file. There was a recent PR (#807) addressing the fact that we wanted to change our assumption on how last names are guessed, so it's something we ought to be consistent about.

Reply:
Great idea, I created `shared/utils/names.js` to address this.